velox/common/memory/MemoryPool.cpp

```diff
@@ -943,4 +952,96 @@ void MemoryPoolImpl::testingSetCapacity(int64_t bytes) {
   std::lock_guard<std::mutex> l(mutex_);
   capacity_ = bytes;
 }
+
+#ifdef VELOX_MEMORY_DEBUG
```

Comment:
Are you talking about the filter API? The filter should be there always. Followup is to add support for dynamic filter. The current usage for this debug code is developer change the filter manually in a static way and then compile the code. Followup will give flexibility to change the filter during run time. Without this filter (static or dynamic) the debug code is not going to work as it will almost always saturate Cpu. I've added TODOs to indicate the future work.